models/BaseEntity.cfc

```diff
@@ -1165,10 +1183,36 @@ component accessors="true" {
 	 * @return  quick.models.BaseEntity
 	 */
 	private any function loadEntity( required struct data ) {
-		return newEntity()
-			.assignAttributesData( arguments.data )
-			.assignOriginalAttributes( arguments.data )
-			.markLoaded();
+		if (
+			variables._loadChildren
+			&&
+			variables._meta.localMetadata.keyExists( "discriminatorColumn" )
+			&&
+			application.quickMeta.discriminators.keyExists( variables._meta.table )
+			&&
+			data.keyExists( variables._meta.localMetadata.discriminatorColumn )
+			&&
+			application.quickMeta.discriminators[ variables._meta.table ].keyExists(
+				data[ variables._meta.localMetadata.discriminatorColumn ]
+			)
+		) {
+			var childClass = variables._wirebox.getInstance(
+				application.quickMeta.discriminators[ variables._meta.table ][
+					data[ variables._meta.localMetadata.discriminatorColumn ]
+				].mapping
+			);
+
+			keyNames().each( function( key, i ) {
+				data[ childClass.keyNames()[ i ] ] = data[ key ];
+			} );
+
+			return childClass.hydrate( data, true );
```

Comment:
One thing missing when using `hydrate` is `assignOriginalAttributes`.  Let's add that in this PR and both paths can use `hydrate`.

Reply:
Done.